xla/literal.h

```diff
@@ -356,16 +356,11 @@ class LiteralBase {
     return ShapeUtil::ElementsIn(ShapeUtil::GetSubshape(shape(), index));
   }
 
-  // This definition is here to ensure that nobody accidentally implements this
-  // function which would lead to inconsistencies. Use Hash instead.
+  // Compute a hash for this literal. Always use this together with the Equal
+  // method and not operator== in order to handle layout sensitivity properly.
   template <typename H>
   friend H AbslHashValue(H state, const LiteralBase& value) {
-    static_assert(false,
-                  "Do not use Literal directly as a hash key, because it has "
-                  "multiple definitions of equality - layout sensitive or "
-                  "insensitive. Instead, provide an external hash function "
-                  "that uses Literal::Hash which allows you to specify layout "
-                  "sensitivity.");
+    return LiteralBase::Hash(std::move(state), value);
```

Comment:
@dimitar-asenov I have this fix https://github.com/openxla/xla/pull/14295 , so maybe no need to revert?